File: tvr-decoder/src/main/java/com/willcurrie/tlv/PrimitiveBerTlv.java
package com.willcurrie.tlv;

import java.util.Collections;
import java.util.List;

import com.willcurrie.EmvTags;

class PrimitiveBerTlv extends BerTlv {

    private final byte[] value;

    PrimitiveBerTlv(Tag tag, byte[] value) {
        super(tag);
        this.value = value;
    }

    public byte[] getValue() {
        return value;
    }

    public BerTlv findTlv(Tag tag) {
        return null;
    }

    public List<BerTlv> findTlvs(Tag tag) {
        return Collections.emptyList();
    }

    public List<BerTlv> getChildren() {
        return Collections.emptyList();
    }

    public String toString() {
        if (EmvTags.isTagAscii(getTag())) {
            return getTag() + ": " + new String(value);
        }
        return getTag() + ": " + ISOUtil.hexString(value);
    }
}

File: src/main/java/io/github/binaryfoo/tlv/PrimitiveBerTlv.kt
package io.github.binaryfoo.tlv

import kotlin.collections.listOf

/**
 * The V is just some bytes.
 */
class PrimitiveBerTlv(tag: Tag, private val value: ByteArray) : BerTlv(tag) {

    override fun getValue(): ByteArray = value

    override fun findTlv(tag: Tag): BerTlv? {
        return null
    }

    override fun findTlvs(tag: Tag): List<BerTlv> {
        return listOf()
    }

    override fun getChildren(): List<BerTlv> {
        return listOf()
    }

    override fun toString(): String = "$tag: ${ISOUtil.hexString(value)}"
}

